mmcv/runner/epoch_based_runner.py

```diff
@@ -161,7 +161,7 @@ def save_checkpoint(self,
             # meta.update(epoch=self.epoch + 1, iter=self.iter) otherwise
             # there will be problems with resumed checkpoints.
             # More details in https://github.com/open-mmlab/mmcv/pull/1108
-        meta.update(epoch=self.epoch + 1, iter=self.iter)
+        meta.update(epoch=self.epoch + 1, iter=self.iter + 1)
```

Comment:
The `EpochBasedRunner.save_checkpoint` is invoked by `CheckpointHook.after_train_epoch`, and at this time, the `self. iter` is a correct iteration number.

Reply:
I think the reason why we need to `+ 1` is the `self._epoch` and `self._iter` update after all the hooks are called, while they will be resumed directly from the checkpoint.

Reply:
Yet, the `self._epoch` will only increment itself by one after all hooks have been called, so we need +1. However, the `self_iter` would be updated before the `EpochBasedRunner.save_checkpoint` is invoking by `CheckpointHook.after_train_epoch`